packages/eslint-plugin/rules/no-store-string-literals.js

```diff
@@ -0,0 +1,139 @@
+function getReferences( context, specifiers ) {
+	const variables = specifiers.reduce(
+		( acc, specifier ) =>
+			acc.concat( context.getDeclaredVariables( specifier ) ),
+		[]
+	);
+	const references = variables.reduce(
+		( acc, variable ) => acc.concat( variable.references ),
+		[]
+	);
+	return references;
+}
+
+function collectAllNodesFromCallbackFunctions( context, node ) {
+	const functionSpecifiers = node.specifiers.filter(
+		( specifier ) =>
+			specifier.imported &&
+			[
+				'createRegistrySelector',
+				'useSelect',
+				'withSelect',
+				'withDispatch',
+			].includes( specifier.imported.name )
+	);
+	const functionReferences = getReferences( context, functionSpecifiers );
+
+	const functionArgumentVariables = functionReferences.reduce(
+		( acc, { identifier: { parent } } ) =>
+			parent && parent.arguments && parent.arguments.length > 0
+				? acc.concat(
+						context.getDeclaredVariables( parent.arguments[ 0 ] )
+				  )
+				: acc,
+		[]
+	);
+	const functionArgumentReferences = functionArgumentVariables.reduce(
+		( acc, variable ) => acc.concat( variable.references ),
+		[]
+	);
+	const possibleCallExpressionNodes = functionArgumentReferences
+		.filter( ( reference ) => reference.identifier.parent )
+		.map( ( reference ) => reference.identifier.parent );
+
+	return possibleCallExpressionNodes;
+}
+
+function collectAllNodesFromDirectFunctionCalls( context, node ) {
+	const specifiers = node.specifiers.filter(
+		( specifier ) =>
+			specifier.imported &&
+			[ 'useDispatch', 'dispatch', 'select', 'resolveSelect' ].includes(
+				specifier.imported.name
+			)
+	);
+	const references = getReferences( context, specifiers );
+	const possibleCallExpressionNodes = references
+		.filter( ( reference ) => reference.identifier.parent )
+		.map( ( reference ) => reference.identifier.parent );
+
+	return possibleCallExpressionNodes;
+}
+
+function collectAllNodesFromObjectPropertyFunctionCalls( context, node ) {
+	const specifiers = node.specifiers.filter(
+		( specifier ) =>
+			specifier.imported &&
+			[ 'controls' ].includes( specifier.imported.name )
+	);
+	const references = getReferences( context, specifiers );
+	const referencesWithPropertyCalls = references.filter(
+		( reference ) =>
+			reference.identifier.parent.property &&
+			[ 'select', 'resolveSelect', 'dispatch' ].includes(
+				reference.identifier.parent.property.name
+			)
+	);
+	const possibleCallExpressionNodes = referencesWithPropertyCalls
+		.filter(
+			( reference ) =>
+				reference.identifier.parent &&
+				reference.identifier.parent.parent
+		)
+		.map( ( reference ) => reference.identifier.parent.parent );
+
+	return possibleCallExpressionNodes;
+}
+
+module.exports = {
+	meta: {
+		type: 'problem',
+		schema: [],
+		messages: {
+			doNotUseStringLiteral: `Do not use string literals ( '{{ argument }}' ) for accessing stores ; import the store and use the store object or store name constant instead`,
```

Comment:
In the majority of cases, it should be the store definition so we could rephrase for simplicity to something like:
```suggestion
			doNotUseStringLiteral: `Do not use string literals ( '{{ argument }}' ) for accessing @wordpress/data stores. Pass the store definition instead`,
```